modules/rostests/apitests/ntdll/RtlCriticalSection.c

```diff
@@ -0,0 +1,374 @@
+        _SEH2_END;
+        ok_ntstatus(Status, STATUS_ACCESS_VIOLATION);
+
+        Status = pfnRtlInitializeCriticalSectionEx(&CritSect, 0, 0);
+        ok_ntstatus(Status, STATUS_SUCCESS);
+        ok_long(CritSect.LockCount, -1);
+        ok_long(CritSect.RecursionCount, 0);
+        ok_ptr(CritSect.OwningThread, NULL);
+        ok_ptr(CritSect.LockSemaphore, NULL);
+        ok_size_t(CritSect.SpinCount, DefaultSpinCount);
+        if (HasDebugInfo)
+        {
+            ok(CritSect.DebugInfo != NULL, "DebugInfo is %p\n", CritSect.DebugInfo);
+            ok(CritSect.DebugInfo != LongToPtr(-1), "DebugInfo is %p\n", CritSect.DebugInfo);
+            if ((CritSect.DebugInfo != NULL) && (CritSect.DebugInfo != LongToPtr(-1)))
+            {
+                ok_int(CritSect.DebugInfo->Type, 0);
+                ok_int(CritSect.DebugInfo->CreatorBackTraceIndex, 0);
+                ok_int(CritSect.DebugInfo->CreatorBackTraceIndexHigh, 0);
+                ok_ptr(CritSect.DebugInfo->CriticalSection, &CritSect);
+                ok(CritSect.DebugInfo->ProcessLocksList.Flink != NULL, "Flink is NULL\n");
+                ok(CritSect.DebugInfo->ProcessLocksList.Blink != NULL, "Blink is NULL\n");
+                if ((CritSect.DebugInfo->ProcessLocksList.Flink != NULL) &&
+                    (CritSect.DebugInfo->ProcessLocksList.Blink != NULL))
+                {
+                    ok_ptr(CritSect.DebugInfo->ProcessLocksList.Flink->Blink,
+                        &CritSect.DebugInfo->ProcessLocksList);
+                    ok_ptr(CritSect.DebugInfo->ProcessLocksList.Blink->Flink,
+                        &CritSect.DebugInfo->ProcessLocksList);
+                }
+                ok_long(CritSect.DebugInfo->EntryCount, 0);
+                ok_long(CritSect.DebugInfo->ContentionCount, 0);
+                ok_long(CritSect.DebugInfo->Flags, 0);
+                ok_int(CritSect.DebugInfo->SpareWORD, 0);
+            }
+        }
+        else
+        {
+            ok(CritSect.DebugInfo == LongToPtr(-1), "DebugInfo is %p\n", CritSect.DebugInfo);
+        }
+
+        Status = pfnRtlInitializeCriticalSectionEx(&CritSect, 0x00FFFFFF, 0);
+        ok_size_t(CritSect.SpinCount, (g_SysInfo.dwNumberOfProcessors > 1) ? 0x00FFFFFF : 0);
+
+        Status = pfnRtlInitializeCriticalSectionEx(&CritSect, 0x01000000, 0);
+        ok_ntstatus(Status, STATUS_INVALID_PARAMETER_2);
+
+        Status = pfnRtlInitializeCriticalSectionEx(&CritSect, 0x80000000, 0);
+        ok_ntstatus(Status, STATUS_INVALID_PARAMETER_2);
+
+        _SEH2_TRY
+        {
+            Status = pfnRtlInitializeCriticalSectionEx(NULL, 0x12345678, 0);
+        }
+        _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
+        {
+            Status = _SEH2_GetExceptionCode();
+        }
+        _SEH2_END;
+        ok_ntstatus(Status, STATUS_INVALID_PARAMETER_2);
+
+        for (ULONG i = 0; i < 32; i++)
+        {
+            ULONG Flags = 1UL << i;
+            ULONG AllowedFlags = 0x07000000;
```

Comment:
Nit: I'd find this more straightforward if you added the `0xFFFFFF` bits here instead of below.